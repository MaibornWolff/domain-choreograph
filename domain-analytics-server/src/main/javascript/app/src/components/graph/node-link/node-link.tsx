import * as React from 'react';
import { Link } from 'react-router-dom';
import { styled } from '~utils/styled';

const StyledLink = styled(Link)`
  position: absolute;
  z-index: 2;
  top: .3rem;
  right: .3rem;
  cursor: pointer;

  &, & > * {
    height: .8rem;
    width: .8rem;
    box-sizing: border-box;
  }

  & > * {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
  }
`;

const InnerPoint = styled('div')`
  background: ${({ theme }) => theme.colors.secondary[0]};
  border-radius: 100%;
  transition: transform 300ms ease;
  transform: scale(.8);

  ${StyledLink as any}:hover & {
    transform: scale(.5);
  }
`;

const OuterPoint = styled('div')`
  background: transparent;
  border: 1px solid ${({ theme }) => theme.colors.secondary[0]};
  border-radius: 100%;
  transition: transform 300ms ease;
  transform: scale(.8);

  ${StyledLink as any}:hover & {
    transform: scale(1);
  }
`;

export interface NodeLinkProps {
  executionContextId: string;
}

export const NodeLink: React.StatelessComponent<NodeLinkProps> = ({ executionContextId }) => (
  <StyledLink to={`/${executionContextId}`} onClick={(event) => event.stopPropagation()}>
    <InnerPoint/>
    <OuterPoint/>
  </StyledLink>
);
