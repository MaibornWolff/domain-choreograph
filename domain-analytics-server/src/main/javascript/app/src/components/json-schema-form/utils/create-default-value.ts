import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';

const defaultValues: Record<Exclude<JSONSchema4TypeName, 'object'>, any> = {
  string: undefined,
  number: undefined,
  integer: undefined,
  boolean: undefined,
  array: [],
  'null': null,
  any: ''
};

export function createDefaultValue(schema: JSONSchema4): any {
  const type = schema.type || 'any';
  if (type instanceof Array) {
    throw new Error('Array types are not supported');
  } else if (type === 'object') {
    return createDefaultValueForObject(schema);
  } else {
    return defaultValues[type];
  }
}

function createDefaultValueForObject(schema: JSONSchema4): Promise<object> {
  if (schema.type !== 'object') {
    throw new Error('Expect schema to be an object');
  }
  let defaultValue: any = {};
  if (schema.properties) {
    Object.entries(schema.properties).forEach(([key, property]) => {
      defaultValue[key] = createDefaultValue(property)
    });
  }
  return defaultValue;
}

