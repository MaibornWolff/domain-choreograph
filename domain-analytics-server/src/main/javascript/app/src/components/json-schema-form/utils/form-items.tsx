import { styled } from '~utils/styled';

export const FormItems = styled('div')`
    display: flex;
    flex-direction: column;

    & > * {
      margin-bottom: .5rem;
    }
`;
