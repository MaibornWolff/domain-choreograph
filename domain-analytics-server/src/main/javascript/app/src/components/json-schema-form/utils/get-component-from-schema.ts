import { FieldComponent } from 'clean-forms';
import { JSONSchema4, JSONSchema4TypeName } from 'json-schema';
import { DatePickerMsField } from '~components/json-schema-form/form-fields/date-picker-ms-field';
import { NumberTextField, TextField } from '~components/json-schema-form/form-fields/text-field';
import { JsonSchemaFormFieldProps } from '~components/json-schema-form/models/json-schema-form-field-props';

export interface JsonSchemaComponentMap {
  matcher?: { type: JSONSchema4TypeName, format?: string };
  field: FieldComponent<any, JsonSchemaFormFieldProps>;
}

const defaultMapping: JsonSchemaComponentMap[] = [
  { matcher: { type: 'integer', format: 'utc-millisec' }, field: DatePickerMsField },
  { matcher: { type: 'integer' }, field: NumberTextField },
  { matcher: { type: 'number' }, field: NumberTextField },
  { field: TextField }
];

export function getComponentFromSchema(
  schema: JSONSchema4,
  mapping: JsonSchemaComponentMap[] = defaultMapping
): FieldComponent<any, JsonSchemaFormFieldProps> {
  for (const map of mapping) {
    if (map.matcher == null) {
      return map.field;
    }
    const typeIsMatching = map.matcher.type === schema.type;
    const formatIsMatching = map.matcher.format == null || map.matcher.format === schema.format;
    if (typeIsMatching && formatIsMatching) {
      return map.field;
    }
  }
  throw new Error('Couldn\'t match the schema');
}
