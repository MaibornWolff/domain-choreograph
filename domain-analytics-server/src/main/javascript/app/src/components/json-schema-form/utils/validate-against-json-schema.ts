import { ValidationFunction } from 'clean-forms';
import { JSONSchema4 } from 'json-schema';
import { Validator } from 'jsonschema';


export function validateAgainstJsonSchema(schema: JSONSchema4): ValidationFunction {
  const validator = new Validator();
  return ({ value }) => {
    const result = validator.validate(value, schema as any);
    const errors = result.errors.map(error => {
      let message = error.message;
      let property = error.property
        .replace(/^instance\.?/, '');
      let schema = error.schema as JSONSchema4;
      if (error.name === 'required') {
        message = `is required`;
        schema = schema.properties![error.argument];
        property = joinProperties(property, error.argument);
      }

      const title = schema.title || '';
      message = `${title} ${message}`;

      return [ property, message ] as [string, string];
    });
    return errors;
  }
}

function joinProperties(path1: string, path2: string): string {
  if (path1 === '') {
    return path2;
  }
  if (path2 === '') {
    return path1;
  }
  return `${path1}.${path2}`;
}
