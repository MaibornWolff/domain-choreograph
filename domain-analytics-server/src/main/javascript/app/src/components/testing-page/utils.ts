export function escapeJavaClass(javaClass: string) {
  return javaClass.replace(/\./g, '-');
}

export function reverseJavaClassEscape(javaClass: string) {
  return javaClass.replace(/-/g, '.');
}
