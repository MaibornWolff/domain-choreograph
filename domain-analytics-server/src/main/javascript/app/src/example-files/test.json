{
  "nodes": {
    "A": {
      "id": "A",
      "name": "A",
      "scope": "root"
    }
  },
  "dependencies": [],
  "scopes": {
    "root": {
      "id": "root",
      "executionContext": "Application",
      "nodes": [
        "A"
      ]
    }
  },
  "executionContexts": {
    "Application": {
      "id": "Application",
      "scopes": [
        "root"
      ]
    }
  }
}
