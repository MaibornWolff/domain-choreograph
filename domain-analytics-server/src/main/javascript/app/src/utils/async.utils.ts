export function wait(milliseconds: number): Promise<void> {
  return new Promise<void>((resolve, reject) => {
    setTimeout(() => resolve(), milliseconds);
  });
}
