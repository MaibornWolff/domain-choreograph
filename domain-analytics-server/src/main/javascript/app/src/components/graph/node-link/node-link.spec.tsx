import * as React from 'react';
import { renderWithProviders } from '~utils/test.utils';
import { NodeLink } from './node-link';

describe('NodeLink', () => {
  it('should render', () => {
    expect(renderWithProviders(<NodeLink executionContextId={'id'}/>)).toMatchSnapshot();
  });
});
