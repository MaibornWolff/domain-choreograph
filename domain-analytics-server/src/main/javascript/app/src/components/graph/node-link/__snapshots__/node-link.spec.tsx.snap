// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeLink should render 1`] = `
.emotion-4 {
  position: absolute;
  z-index: 2;
  top: .3rem;
  right: .3rem;
  cursor: pointer;
}

.emotion-4,
.emotion-4 > * {
  height: .8rem;
  width: .8rem;
  box-sizing: border-box;
}

.emotion-4 > * {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.emotion-0 {
  background: #39ac60;
  border-radius: 100%;
  -webkit-transition: -webkit-transform 300ms ease;
  -webkit-transition: transform 300ms ease;
  transition: transform 300ms ease;
  -webkit-transform: scale(.8);
  -ms-transform: scale(.8);
  transform: scale(.8);
}

.emotion-5:hover .emotion-0 {
  -webkit-transform: scale(.5);
  -ms-transform: scale(.5);
  transform: scale(.5);
}

.emotion-2 {
  background: transparent;
  border: 1px solid #39ac60;
  border-radius: 100%;
  -webkit-transition: -webkit-transform 300ms ease;
  -webkit-transition: transform 300ms ease;
  transition: transform 300ms ease;
  -webkit-transform: scale(.8);
  -ms-transform: scale(.8);
  transform: scale(.8);
}

.emotion-5:hover .emotion-2 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<a
  class="emotion-4 emotion-5"
  href="/id"
>
  <div
    class="emotion-0 emotion-1"
  />
  <div
    class="emotion-2 emotion-3"
  />
</a>
`;
